# Індивідуальне завдання № 2

## *Порівняння кількості теплоти необхідної для ізобаричного та ізохоричного нагріву суміші*

## Формулювання завдання:

### Вихідні дані

Розглядається газова суміш, склад якої змінюється покроково під час нагрівання.

Температура та склад суміші на кожному кроці наведено в таблиці 1.

#### Таблиця 1. Температура та склад газової суміші на кожному кроці (об'ємні %).

| Крок | Температура (°C) | CO (%) | CO₂ (%) | O₂ (%) | N₂ (%) |
|------|------------------|--------|---------|--------|--------|
| 1    | 25               | 87.4   | 10.1    | 1.6    | 0.9    |
| 2    | 114              | 87.1   | 10.6    | 1.45   | 0.85   |
| 3    | 202              | 82.7   | 15.2    | 1.3    | 0.8    |
| 4    | 291              | 77.6   | 20.6    | 1.1    | 0.7    |
| 5    | 380              | 73.5   | 25.0    | 0.9    | 0.6    |
| 6    | 468              | 67.3   | 31.5    | 0.7    | 0.5    |
| 7    | 557              | 55.5   | 43.5    | 0.6    | 0.4    |
| 8    | 645              | 40.1   | 59.1    | 0.5    | 0.3    |
| 9    | 734              | 25.9   | 73.4    | 0.4    | 0.3    |
| 10   | 823              | 20.3   | 79.2    | 0.3    | 0.2    |
| 11   | 911              | 12.5   | 87.1    | 0.2    | 0.2    |
| 12   | 1000             | 8.8    | 91.0    | 0.1    | 0.1    |

#### Таблиця 2. Коефіцієнти для розрахунку теплоємності газів

Теплоємності деяких газів $c_p = a + bT + cT^{-2}$

| Газ | a | b·10³ | c·10⁻⁵ | ΔT, К |
|-----|-----|--------|---------|--------|
| CO | 6.79 | 0.98 | -0.11 | 298-2500 |
| CO₂ | 10.55 | 2.16 | -2.04 | 298-2500 |
| CH₄ | 5.65 | 11.44 | -0.46 | 298-1500 |
| H₂ | 27.30 | 3.27 | 0.50 | 298-3000 |
| N₂ | 6.66 | 1.02 | - | 298-2500 |
| O₂ | 7.16 | 1.0 | -0.40 | 298-3000 |

*Примітка: Коефіцієнти наведені в кал/(моль·К)*

### Умова задачі
Кожен студент обирає крок відповідно своєму варіанту та розраховує необхідну кількість теплоти
для нагріву суміші до температури наступного кроку при умови постійного тиску та постійного об'єму.

*Прийняти для кроку*: початковий об'єм $100$ м³, початковий тиск $101325$ Па.

Для *ізохоричного* нагріву прийняти постійну кількість речовини.

Для *ізобаричного* нагріву постійний тиск підтримується шляхом зменшення кількості речовини.

Відповіддю має бути порівняння кількостей тепла для цих двох випадків.

# Розв'язок індивідуального завдання №2 (Варіант 11)

### Вихідні дані

Відповідно до варіанту 11, розглядаємо нагрівання газової суміші від кроку 11 до кроку 12:
- Початкова температура: 911°C
- Кінцева температура: 1000°C
- Початковий склад суміші (об'ємні %):
  - CO: 12.5%
  - CO₂: 87.1%
  - O₂: 0.2%
  - N₂: 0.2%
- Кінцевий склад суміші (об'ємні %):
  - CO: 8.8%
  - CO₂: 91.0%
  - O₂: 0.1%
  - N₂: 0.1%
- Початковий об'єм: 100 м³
- Початковий тиск: 101325 Па

### Коефіцієнти для розрахунку теплоємності газів

| Газ | a | b·10³ | c·10⁻⁵ | ΔT, К |
|-----|-----|--------|---------|--------|
| CO | 6.79 | 0.98 | -0.11 | 298-2500 |
| CO₂ | 10.55 | 2.16 | -2.04 | 298-2500 |
| N₂ | 6.66 | 1.02 | - | 298-2500 |
| O₂ | 7.16 | 1.0 | -0.40 | 298-3000 |

*Примітка: Коефіцієнти наведені в кал/(моль·К)*


## Методика розв'язання

### Крок 1: Розрахунок початкової кількості речовини

Формула для розрахунку кількості речовини

$$ n = \frac{PV}{RT} $$

де:
- $P$ - тиск газу, Па (101325 Па)
- $V$ - об'єм газу, м³ (100 м³)
- $R$ - універсальна газова стала, Дж/(моль·К) (8.31446261815 Дж/(моль·К))
- $T$ - температура газу, К (911°C = 1184.15 К)

In [1]:
# Константи
P = 101325  # Па
V = 100  # м³
R = 8.31446261815  # Дж/(моль·К)
T_C_start = 0  # °C
T_K_start = T_C_start + 273.15  # К

# Розрахунок кількості речовини
n_start = (P * V) / (R * T_K_start)
print(f"Початкова кількість речовини: {n_start:.2f} моль")

Початкова кількість речовини: 4461.50 моль


Результат розрахунку

$$ n = \frac{101325 \cdot 100}{8.31446261815 \cdot 1184.15} = 1029.14 \text{ моль} $$

Початкова кількість речовини: 1029.14 моль

### Крок 2: Розрахунок кількості речовини кожного компонента

Формула для розрахунку кількості речовини компонентів

$$ n_i = r_i \cdot n $$

де:
- $n_i$ - кількість речовини i-го компонента, моль
- $r_i$ - об'ємна частка i-го компонента (в долях одиниці)
- $n$ - загальна кількість речовини, моль (1029.14 моль)

In [2]:
# Об'ємні частки компонентів (початкові)
r_co_start = 0.125  # 12.5%
r_co2_start = 0.871  # 87.1%
r_o2_start = 0.002  # 0.2%
r_n2_start = 0.002  # 0.2%

# Розрахунок кількості речовини кожного компонента
n_co_start = r_co_start * n_start
n_co2_start = r_co2_start * n_start
n_o2_start = r_o2_start * n_start
n_n2_start = r_n2_start * n_start

print(f"Початкова кількість CO: {n_co_start:.2f} моль")
print(f"Початкова кількість CO₂: {n_co2_start:.2f} моль")
print(f"Початкова кількість O₂: {n_o2_start:.2f} моль")
print(f"Початкова кількість N₂: {n_n2_start:.2f} моль")

Початкова кількість CO: 557.69 моль
Початкова кількість CO₂: 3885.97 моль
Початкова кількість O₂: 8.92 моль
Початкова кількість N₂: 8.92 моль


Результат розрахунку

$$ n_{CO} = 0.125 \cdot 1029.14 = 128.64 \text{ моль} $$
$$ n_{CO_2} = 0.871 \cdot 1029.14 = 896.38 \text{ моль} $$
$$ n_{O_2} = 0.002 \cdot 1029.14 = 2.06 \text{ моль} $$
$$ n_{N_2} = 0.002 \cdot 1029.14 = 2.06 \text{ моль} $$

Початкова кількість CO: 128.64 моль

Початкова кількість CO₂: 896.38 моль

Початкова кількість O₂: 2.06 моль

Початкова кількість N₂: 2.06 моль

### Крок 3: Розрахунок коефіцієнтів теплоємності при постійному тиску

Формула для розрахунку теплоємності при постійному тиску

$$ c_p = a + bT + cT^{-2} \text{ кал/(моль·К)} $$

де:
- $a$, $b$, $c$ - коефіцієнти з таблиці
- $T$ - температура, К

Переведення в Дж/(моль·К):
$$ c_p[\text{Дж/(моль·К)}] = c_p[\text{кал/(моль·К)}] \cdot 4.1868 $$

In [3]:
# Коефіцієнти теплоємності (кал/(моль·К))
# CO
a_co = 6.79
b_co = 0.98e-3
c_co = -0.11e5
# CO₂
a_co2 = 10.55
b_co2 = 2.16e-3
c_co2 = -2.04e5
# O₂
a_o2 = 7.16
b_o2 = 1.0e-3
c_o2 = -0.40e5
# N₂
a_n2 = 6.66
b_n2 = 1.02e-3
c_n2 = 0

# Коефіцієнт переведення з кал в Дж
CAL_TO_JOULE = 4.1868

# Середня температура для розрахунку (К)
T_C_end = 1000  # °C
T_K_end = T_C_end + 273.15  # К
T_avg = (T_K_start + T_K_end) / 2

# Розрахунок теплоємності при постійному тиску (Дж/(моль·К))
cp_co = (a_co + b_co * T_avg + c_co * T_avg**(-2)) * CAL_TO_JOULE
cp_co2 = (a_co2 + b_co2 * T_avg + c_co2 * T_avg**(-2)) * CAL_TO_JOULE
cp_o2 = (a_o2 + b_o2 * T_avg + c_o2 * T_avg**(-2)) * CAL_TO_JOULE
cp_n2 = (a_n2 + b_n2 * T_avg) * CAL_TO_JOULE

print(f"Теплоємність CO при постійному тиску: {cp_co:.2f} Дж/(моль·К)")
print(f"Теплоємність CO₂ при постійному тиску: {cp_co2:.2f} Дж/(моль·К)")
print(f"Теплоємність O₂ при постійному тиску: {cp_o2:.2f} Дж/(моль·К)")
print(f"Теплоємність N₂ при постійному тиску: {cp_n2:.2f} Дж/(моль·К)")

Теплоємність CO при постійному тиску: 31.52 Дж/(моль·К)
Теплоємність CO₂ при постійному тиску: 49.73 Дж/(моль·К)
Теплоємність O₂ при постійному тиску: 32.93 Дж/(моль·К)
Теплоємність N₂ при постійному тиску: 31.19 Дж/(моль·К)


Результат розрахунку

Середня температура: $T_{avg} = (1184.15 + 1273.15) / 2 = 1228.65$ К

$$ c_p^{CO} = (6.79 + 0.98 \cdot 10^{-3} \cdot 1228.65 - 0.11 \cdot 10^{5} \cdot 1228.65^{-2}) \cdot 4.1868 = 33.44 \text{ Дж/(моль·К)} $$

$$ c_p^{CO_2} = (10.55 + 2.16 \cdot 10^{-3} \cdot 1228.65 - 2.04 \cdot 10^{5} \cdot 1228.65^{-2}) \cdot 4.1868 = 54.72 \text{ Дж/(моль·К)} $$

$$ c_p^{O_2} = (7.16 + 1.0 \cdot 10^{-3} \cdot 1228.65 - 0.40 \cdot 10^{5} \cdot 1228.65^{-2}) \cdot 4.1868 = 35.01 \text{ Дж/(моль·К)} $$

$$ c_p^{N_2} = (6.66 + 1.02 \cdot 10^{-3} \cdot 1228.65) \cdot 4.1868 = 33.13 \text{ Дж/(моль·К)} $$

Теплоємність CO при постійному тиску: 33.44 Дж/(моль·К)

Теплоємність CO₂ при постійному тиску: 54.72 Дж/(моль·К)

Теплоємність O₂ при постійному тиску: 35.01 Дж/(моль·К)

Теплоємність N₂ при постійному тиску: 33.13 Дж/(моль·К)

### Крок 4: Розрахунок коефіцієнтів теплоємності при постійному об'ємі

Формула для розрахунку теплоємності при постійному об'ємі

$$ c_v = c_p - R $$

де:
- $c_p$ - теплоємність при постійному тиску, Дж/(моль·К)
- $R$ - універсальна газова стала, Дж/(моль·К) (8.31446261815 Дж/(моль·К))

In [4]:
# Розрахунок теплоємності при постійному об'ємі (Дж/(моль·К))
cv_co = cp_co - R
cv_co2 = cp_co2 - R
cv_o2 = cp_o2 - R
cv_n2 = cp_n2 - R

print(f"Теплоємність CO при постійному об'ємі: {cv_co:.2f} Дж/(моль·К)")
print(f"Теплоємність CO₂ при постійному об'ємі: {cv_co2:.2f} Дж/(моль·К)")
print(f"Теплоємність O₂ при постійному об'ємі: {cv_o2:.2f} Дж/(моль·К)")
print(f"Теплоємність N₂ при постійному об'ємі: {cv_n2:.2f} Дж/(моль·К)")

Теплоємність CO при постійному об'ємі: 23.21 Дж/(моль·К)
Теплоємність CO₂ при постійному об'ємі: 41.42 Дж/(моль·К)
Теплоємність O₂ при постійному об'ємі: 24.62 Дж/(моль·К)
Теплоємність N₂ при постійному об'ємі: 22.87 Дж/(моль·К)


Результат розрахунку

$$ c_v^{CO} = 33.44 - 8.314 = 25.12 \text{ Дж/(моль·К)} $$
$$ c_v^{CO_2} = 54.72 - 8.314 = 46.40 \text{ Дж/(моль·К)} $$
$$ c_v^{O_2} = 35.01 - 8.314 = 26.70 \text{ Дж/(моль·К)} $$
$$ c_v^{N_2} = 33.13 - 8.314 = 24.82 \text{ Дж/(моль·К)} $$

Теплоємність CO при постійному об'ємі: 25.12 Дж/(моль·К)

Теплоємність CO₂ при постійному об'ємі: 46.40 Дж/(моль·К)

Теплоємність O₂ при постійному об'ємі: 26.70 Дж/(моль·К)

Теплоємність N₂ при постійному об'ємі: 24.82 Дж/(моль·К)

### Крок 5: Розрахунок теплоти нагрівання при постійному об'ємі (ізохоричний процес)

Формула для розрахунку теплоти нагрівання при постійному об'ємі

$$ Q_v = \sum_{i} n_i \cdot c_{v,i} \cdot (T_2 - T_1) $$

де:
- $n_i$ - кількість речовини i-го компонента, моль
- $c_{v,i}$ - теплоємність i-го компонента при постійному об'ємі, Дж/(моль·К)
- $T_2$ - кінцева температура, К (1000°C = 1273.15 К)
- $T_1$ - початкова температура, К (911°C = 1184.15 К)

In [5]:
# Різниця температур (К)
delta_T = T_K_end - T_K_start

# Розрахунок теплоти нагрівання при постійному об'ємі (Дж)
Q_v_co = n_co_start * cv_co * delta_T
Q_v_co2 = n_co2_start * cv_co2 * delta_T
Q_v_o2 = n_o2_start * cv_o2 * delta_T
Q_v_n2 = n_n2_start * cv_n2 * delta_T

Q_v_total = Q_v_co + Q_v_co2 + Q_v_o2 + Q_v_n2

print(f"Теплота нагрівання CO при постійному об'ємі: {Q_v_co/1000:.2f} кДж")
print(f"Теплота нагрівання CO₂ при постійному об'ємі: {Q_v_co2/1000:.2f} кДж")
print(f"Теплота нагрівання O₂ при постійному об'ємі: {Q_v_o2/1000:.2f} кДж")
print(f"Теплота нагрівання N₂ при постійному об'ємі: {Q_v_n2/1000:.2f} кДж")
print(f"Загальна теплота нагрівання при постійному об'ємі: {Q_v_total/1000:.2f} кДж")

Теплота нагрівання CO при постійному об'ємі: 12943.46 кДж
Теплота нагрівання CO₂ при постійному об'ємі: 160954.54 кДж
Теплота нагрівання O₂ при постійному об'ємі: 219.68 кДж
Теплота нагрівання N₂ при постійному об'ємі: 204.08 кДж
Загальна теплота нагрівання при постійному об'ємі: 174321.77 кДж


Результат розрахунку

$$ \Delta T = 1273.15 - 1184.15 = 89 \text{ К} $$

$$ Q_{v,CO} = 128.64 \cdot 25.12 \cdot 89 = 287.66 \text{ кДж} $$
$$ Q_{v,CO_2} = 896.38 \cdot 46.40 \cdot 89 = 3701.85 \text{ кДж} $$
$$ Q_{v,O_2} = 2.06 \cdot 26.70 \cdot 89 = 4.89 \text{ кДж} $$
$$ Q_{v,N_2} = 2.06 \cdot 24.82 \cdot 89 = 4.55 \text{ кДж} $$

$$ Q_{v,total} = 287.66 + 3701.85 + 4.89 + 4.55 = 3998.94 \text{ кДж} $$

Теплота нагрівання CO при постійному об'ємі: 287.66 кДж

Теплота нагрівання CO₂ при постійному об'ємі: 3701.85 кДж

Теплота нагрівання O₂ при постійному об'ємі: 4.89 кДж

Теплота нагрівання N₂ при постійному об'ємі: 4.55 кДж

Загальна теплота нагрівання при постійному об'ємі: 3998.94 кДж

### Крок 6: Розрахунок кінцевої кількості речовини для ізобаричного процесу

Формула для розрахунку кінцевої кількості речовини при постійному тиску

$$ n_2 = n_1 \cdot \frac{T_1}{T_2} $$

де:
- $n_2$ - кінцева кількість речовини, моль
- $n_1$ - початкова кількість речовини, моль (1029.14 моль)
- $T_1$ - початкова температура, К (911°C = 1184.15 К)
- $T_2$ - кінцева температура, К (1000°C = 1273.15 К)


In [6]:
# Розрахунок кінцевої кількості речовини для ізобаричного процесу
n_end_isobaric = n_start * (T_K_start / T_K_end)

# Розрахунок кінцевої кількості речовини кожного компонента
# Використовуємо кінцеві об'ємні частки
r_co_end = 0.088  # 8.8%
r_co2_end = 0.910  # 91.0%
r_o2_end = 0.001  # 0.1%
r_n2_end = 0.001  # 0.1%

n_co_end = r_co_end * n_end_isobaric
n_co2_end = r_co2_end * n_end_isobaric
n_o2_end = r_o2_end * n_end_isobaric
n_n2_end = r_n2_end * n_end_isobaric

print(f"Кінцева кількість речовини (ізобаричний процес): {n_end_isobaric:.2f} моль")
print(f"Кінцева кількість CO: {n_co_end:.2f} моль")
print(f"Кінцева кількість CO₂: {n_co2_end:.2f} моль")
print(f"Кінцева кількість O₂: {n_o2_end:.2f} моль")
print(f"Кінцева кількість N₂: {n_n2_end:.2f} моль")

Кінцева кількість речовини (ізобаричний процес): 957.20 моль
Кінцева кількість CO: 84.23 моль
Кінцева кількість CO₂: 871.05 моль
Кінцева кількість O₂: 0.96 моль
Кінцева кількість N₂: 0.96 моль


Результат розрахунку

$$ n_2 = 1029.14 \cdot \frac{1184.15}{1273.15} = 957.20 \text{ моль} $$

$$ n_{CO,end} = 0.088 \cdot 957.20 = 84.23 \text{ моль} $$
$$ n_{CO_2,end} = 0.910 \cdot 957.20 = 871.05 \text{ моль} $$
$$ n_{O_2,end} = 0.001 \cdot 957.20 = 0.96 \text{ моль} $$
$$ n_{N_2,end} = 0.001 \cdot 957.20 = 0.96 \text{ моль} $$

Кінцева кількість речовини (ізобаричний процес): 957.20 моль

Кінцева кількість CO: 84.23 моль

Кінцева кількість CO₂: 871.05 моль

Кінцева кількість O₂: 0.96 моль

Кінцева кількість N₂: 0.96 моль

### Крок 7: Розрахунок теплоти нагрівання при постійному тиску (ізобаричний процес)

#### 7.1 Формула для розрахунку теплоти нагрівання при постійному тиску

$$ Q_p = \sum_{i} \bar{n}_i \cdot c_{p,i} \cdot (T_2 - T_1) $$

де:
- $\bar{n}_i$ - середня кількість речовини i-го компонента, моль
- $c_{p,i}$ - теплоємність i-го компонента при постійному тиску, Дж/(моль·К)
- $T_2$ - кінцева температура, К (1000°C = 1273.15 К)
- $T_1$ - початкова температура, К (911°C = 1184.15 К)

In [7]:
# Розрахунок середньої кількості речовини для кожного компонента
n_co_avg = (n_co_start + n_co_end) / 2
n_co2_avg = (n_co2_start + n_co2_end) / 2
n_o2_avg = (n_o2_start + n_o2_end) / 2
n_n2_avg = (n_n2_start + n_n2_end) / 2

# Розрахунок теплоти нагрівання при постійному тиску (Дж)
Q_p_co = n_co_avg * cp_co * delta_T
Q_p_co2 = n_co2_avg * cp_co2 * delta_T
Q_p_o2 = n_o2_avg * cp_o2 * delta_T
Q_p_n2 = n_n2_avg * cp_n2 * delta_T

Q_p_total = Q_p_co + Q_p_co2 + Q_p_o2 + Q_p_n2

print(f"Теплота нагрівання CO при постійному тиску: {Q_p_co/1000:.2f} кДж")
print(f"Теплота нагрівання CO₂ при постійному тиску: {Q_p_co2/1000:.2f} кДж")
print(f"Теплота нагрівання O₂ при постійному тиску: {Q_p_o2/1000:.2f} кДж")
print(f"Теплота нагрівання N₂ при постійному тиску: {Q_p_n2/1000:.2f} кДж")
print(f"Загальна теплота нагрівання при постійному тиску: {Q_p_total/1000:.2f} кДж")

Теплота нагрівання CO при постійному тиску: 10117.84 кДж
Теплота нагрівання CO₂ при постійному тиску: 118292.55 кДж
Теплота нагрівання O₂ при постійному тиску: 162.70 кДж
Теплота нагрівання N₂ при постійному тиску: 154.06 кДж
Загальна теплота нагрівання при постійному тиску: 128727.15 кДж


Результат розрахунку

$$ \bar{n}_{CO} = \frac{128.64 + 84.23}{2} = 106.44 \text{ моль} $$
$$ \bar{n}_{CO_2} = \frac{896.38 + 871.05}{2} = 883.72 \text{ моль} $$
$$ \bar{n}_{O_2} = \frac{2.06 + 0.96}{2} = 1.51 \text{ моль} $$
$$ \bar{n}_{N_2} = \frac{2.06 + 0.96}{2} = 1.51 \text{ моль} $$

$$ Q_{p,CO} = 106.44 \cdot 33.44 \cdot 89 = 316.77 \text{ кДж} $$
$$ Q_{p,CO_2} = 883.72 \cdot 54.72 \cdot 89 = 4303.48 \text{ кДж} $$
$$ Q_{p,O_2} = 1.51 \cdot 35.01 \cdot 89 = 4.70 \text{ кДж} $$
$$ Q_{p,N_2} = 1.51 \cdot 33.13 \cdot 89 = 4.45 \text{ кДж} $$

$$ Q_{p,total} = 316.77 + 4303.48 + 4.70 + 4.45 = 4629.39 \text{ кДж} $$

Теплота нагрівання CO при постійному тиску: 316.77 кДж

Теплота нагрівання CO₂ при постійному тиску: 4303.48 кДж

Теплота нагрівання O₂ при постійному тиску: 4.70 кДж

Теплота нагрівання N₂ при постійному тиску: 4.45 кДж

Загальна теплота нагрівання при постійному тиску: 4629.39 кДж

## Порівняння результатів

### Таблиця порівняння

| Параметр | Ізохоричний процес | Ізобаричний процес | Різниця |
|----------|-------------------|-------------------|---------|
| Теплота нагрівання | 3998.94 кДж | 4629.39 кДж | 630.45 кДж |
| Початкова кількість речовини | 1029.14 моль | 1029.14 моль | 0 моль |
| Кінцева кількість речовини | 1029.14 моль | 957.20 моль | 71.94 моль |

### Висновки

1. Для нагрівання газової суміші від 911°C до 1000°C при постійному об'ємі (ізохоричний процес) необхідно 3998.94 кДж теплоти.
2. Для нагрівання тієї ж суміші при постійному тиску (ізобаричний процес) необхідно 4629.39 кДж теплоти.
3. Різниця між кількістю теплоти для ізобаричного та ізохоричного процесів становить 630.45 кДж, що складає близько 15.8% від теплоти ізохоричного процесу.
4. Ця різниця пояснюється тим, що при ізобаричному процесі частина теплоти витрачається на роботу розширення газу.
5. Для підтримання постійного тиску при нагріванні, кількість речовини в системі зменшилася з 1029.14 моль до 957.20 моль, тобто на 71.94 моль (приблизно 7%).

Таким чином, ізобаричний процес нагрівання потребує більше теплоти, ніж ізохоричний, що відповідає фундаментальним принципам термодинаміки.